pkg/cmd/server/server.go

```diff
@@ -1081,13 +1081,13 @@ func markInProgressBackupsFailed(ctx context.Context, client ctrlclient.Client,
 	}
 
 	for i, backup := range backups.Items {
-		if backup.Status.Phase != velerov1api.BackupPhaseInProgress && backup.Status.Phase != velerov1api.BackupPhaseWaitingForPluginOperations {
```

Comment:
For data mover,   the backup CRD  will be in the `BackupPhaseWaitingForPluginOperations` phase when the `node-agent ` uploads files, assuming that the node-agent pod restarted at this moment  Coincidentally, if we don't mark CRDs failed, the backup would be stuck in `InProgress`

Reply:
I don't think I fully understood the scenario.
can you try to elaborate.

Reply:
Backup can't be stuck `InProgress` if it's already in `WaitingForPluginOperations`. If the velero pod restarts while `WaitingForPluginOperations`, there's no need to fail it. On next reconcile, it will call `Progress` on any in-progress item operations, including DataUpload operations. Since that's handled in the node agent, they aren't interrupted by velero pod restart, so the backup should be able to move to `Completed` as soon as uploads comlete.

Reply:
Sorry, you are right. this is the logic in the Velero server pod but not the `node-agent`, this adjustment of the code logic is correct

Reply:
resolving